solutions/swb-app/src/userRoutes.ts

```diff
@@ -54,15 +55,41 @@ export function setUpUserRoutes(router: Router, user: UserManagementService): vo
         await user.deleteUser(userId);
         res.status(204).send();
       } catch (err) {
-        if (err instanceof UserNotFoundError) {
-          throw Boom.notFound(`Could not find user ${userId}`);
-        }
+        if (err instanceof UserNotFoundError) throw Boom.notFound(`Could not find user ${userId}`);
+        if (Boom.isBoom(err)) throw err;
+        throw Boom.badImplementation(`Could not delete user ${userId}`);
+      }
+    })
+  );
+
+  router.patch(
+    '/users/:userId',
+    wrapAsync(async (req: Request, res: Response) => {
+      processValidatorResult(validate(req.body, UpdateUserSchema));
 
-        if (Boom.isBoom(err)) {
-          throw err;
+      // Since service requires User object, uid attribute has to be added
+      req.body.uid = req.params.userId;
+
+      // Schema verifies status has to be one of these: ACTIVE|INACTIVE
+      req.body.status = req.body.status === 'ACTIVE' ? Status.ACTIVE : Status.INACTIVE;
```

Comment:
We might accidentally deactivate user if status is not sent.

`req.body.status = req.body.status ? undefined :  req.body.status === 'ACTIVE' ? Status.ACTIVE : Status.INACTIVE`

and then check `undefined` later (L79)

Reply:
Good point. Updating..